integration-cli/docker_api_logs_test.go

```diff
@@ -85,3 +86,94 @@ func (s *DockerSuite) TestLogsAPIContainerNotFound(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, http.StatusNotFound)
 }
+
+func extractBody(c *check.C, url string) []string {
+	res, body, err := request.Get(url)
+	defer body.Close()
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	reader := bufio.NewReader(body)
+	bytes, err := ioutil.ReadAll(reader)
+	c.Assert(err, checker.IsNil)
+
+	var out []string
+	for _, line := range strings.Split(string(bytes), "\n") {
+		parts := strings.Split(line, "$")
+		if len(parts) > 1 {
+			out = append(out, parts[1])
+		}
+	}
+	return out
+}
+
+func (s *DockerSuite) TestLogsAPIUntilFutureFollow(c *check.C) {
+	// TODO Windows TP5 - Figure out why this test is so flakey. Disabled for now.
+	// Also disabled in https://github.com/moby/moby/blob/eeaa6c96d83575da765dfc626c2c73d8a29dda32/integration-cli/docker_cli_logs_test.go#L182-L183
+	testRequires(c, DaemonIsLinux)
+
+	until := time.Now().Unix() + 5
+	out, _ := dockerCmd(c, "run", "-d", "-t", "busybox", "/bin/sh", "-c", "while true; do echo hello; sleep 1; done")
+	id := strings.TrimSpace(out)
+	c.Assert(waitRun(id), checker.IsNil)
+
+	type logOut struct {
+		out string
+		res *http.Response
+		err error
+	}
+	chLog := make(chan logOut)
+
+	go func() {
+		res, body, err := request.Get(fmt.Sprintf("/containers/%s/logs?until=%d&follow=1&stdout=1&timestamps=1", id, until))
```

Comment:
I'm not sure; when I tried, it resulted in a lot of obtuse errors. Plus I copied this from a test in this file, so I assume there was a good reason.